File: dotnet/Lexy.Poc/Lexy.Poc.Core/Language/ScenarioParameters.cs
using System.Collections.Generic;
using Lexy.Poc.Core.Parser;

namespace Lexy.Poc.Core.Language
{
    public class ScenarioParameters : ParsableNode
    {
        public IList<AssignmentDefinition> Assignments { get; } = new List<AssignmentDefinition>();

        public override IParsableNode Parse(IParserContext context)
        {
            var assignment = AssignmentDefinition.Parse(context);
            Assignments.Add(assignment);
            return this;
        }

        protected override IEnumerable<INode> GetChildren() => Assignments;

        protected override void Validate(IParserContext context)
        {
        }
    }
}

File: typescipt/existing/src/language/scenarios/scenarioParameters.ts



namespace Lexy.Compiler.Language.Scenarios;

public class ScenarioParameters : ParsableNode
{
   public IList<AssignmentDefinition> Assignments { get; } = new List<AssignmentDefinition>();

   public ScenarioParameters(SourceReference reference) : base(reference)
   {
   }

   public override IParsableNode Parse(IParseLineContext context)
   {
     var assignment = AssignmentDefinition.Parse(context);
     if (assignment ! null) Assignments.Add(assignment);
     return this;
   }

   public override IEnumerable<INode> GetChildren()
   {
     return Assignments;
   }

   protected override void Validate(IValidationContext context)
   {
   }
}
